internal/httpserve/handlers/register.go

```diff
@@ -45,6 +47,24 @@ func (h *Handler) RegisterHandler(ctx echo.Context) error {
 	// set viewer context
 	ctxWithToken := token.NewContextWithSignUpToken(ctx.Request().Context(), in.Email)
 
+	if in.Token != nil {
+		ctxWithToken = token.NewContextWithOrgInviteToken(ctxWithToken, *in.Token)
+
+		invitedUser, err := h.getUserByInviteToken(ctxWithToken, *in.Token)
+		if err != nil {
+			log.Error().Err(err).Msg("error retrieving invite token")
+			return h.BadRequest(ctx, err)
+		}
+
+		if !strings.EqualFold(strings.ToLower(invitedUser.Recipient), strings.ToLower(input.Email)) {
```

Comment:
you shouldn't need `strings.ToLower(` when using `.EqualFold`:

```suggestion
		if !strings.EqualFold(invitedUser.Recipient, input.Email) {
```